Python tests don't work with latest python-bitcoinlib (0.9.0)

People may run into this eventually if they try to run the tests outside of docker:
```
lightningd/test/run-find_my_path
make pytest
make[1]: Entering directory '/home/jb55/dev/github/ElementsProject/lightning'
PYTHONPATH+=:contrib/pylightning DEVELOPER=1 python3 tests/test_lightningd.py -f
Testing results are in /run/user/1000/lightning-hn3fxbx1
ERROR

======================================================================
ERROR: setUpModule (__main__)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_lightningd.py", line 75, in setUpModule
    setupBitcoind()
  File "tests/test_lightningd.py", line 42, in setupBitcoind
    info = bitcoind.rpc.getblockchaininfo()
  File "/home/jb55/dev/github/ElementsProject/lightning/tests/utils.py", line 182, in <lambda>
    f = lambda *args: BitcoinProxy(self.url)._call(name, *args)
  File "/nix/store/iw042323snmxx196lkh4w49xfz7qvj7c-python3.6-bitcoinlib-0.9.0/lib/python3.6/site-packages/bitcoin/rpc.py", line 219, in _call
    response = self._get_response()
  File "/nix/store/iw042323snmxx196lkh4w49xfz7qvj7c-python3.6-bitcoinlib-0.9.0/lib/python3.6/site-packages/bitcoin/rpc.py", line 250, in _get_response
    parse_float=decimal.Decimal)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/json/__init__.py", line 367, in loads
    return cls(**kw).decode(s)
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/nix/store/nrv3zzw9a42vax72gaif8bjgxgnv9iy1-python3-3.6.3/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

----------------------------------------------------------------------
Ran 0 tests in 0.174s

FAILED (errors=1)
make[1]: *** [Makefile:177: pytest] Error 1
make[1]: Leaving directory '/home/jb55/dev/github/ElementsProject/lightning'
make: *** [Makefile:173: check] Error 2

```

--- a/tests/test_lightningd.py
+++ b/tests/test_lightningd.py
@@ -1876,7 +1876,7 @@ class LightningDTests(BaseLightningDTests):
         # Just to be sure, second openingd hand over to channeld.
         l2.daemon.wait_for_log('lightning_openingd.*REPLY WIRE_OPENING_FUNDEE_REPLY with 2 fds')
 
-    @unittest.skipIf(os.getenv("ARCH", "64") == "32" , "temporarily disabled due to flaky behavior, issue #468")
+    @unittest.skip("temporarily disabled due to flaky behavior, issue #468")
     @unittest.skipIf(not DEVELOPER, "needs DEVELOPER=1")
     def test_reconnect_normal(self):
         # Should reconnect fine even if locked message gets lost.

--- a/tests/test_lightningd.py
+++ b/tests/test_lightningd.py
@@ -1876,6 +1876,7 @@ class LightningDTests(BaseLightningDTests):
         # Just to be sure, second openingd hand over to channeld.
         l2.daemon.wait_for_log('lightning_openingd.*REPLY WIRE_OPENING_FUNDEE_REPLY with 2 fds')
 
+    @unittest.skipIf(os.getenv("ARCH", "64") == "32" , "temporarily disabled due to flaky behavior, issue #468")
     @unittest.skipIf(not DEVELOPER, "needs DEVELOPER=1")
     def test_reconnect_normal(self):
         # Should reconnect fine even if locked message gets lost.

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -162,6 +162,7 @@ class TailableProc(object):
         """
         return self.wait_for_logs([regex], timeout)
 
+
 class SimpleBitcoinProxy:
     """Wrapper for BitcoinProxy to reconnect.
 
@@ -170,8 +171,8 @@ class SimpleBitcoinProxy:
     throwaway connections. This is easier than to reach into the RPC
     library to close, reopen and reauth upon failure.
     """
-    def __init__(self, url):
-        self.url = url
+    def __init__(self, btc_conf_file, *args, **kwargs):
+        self.__btc_conf_file__= btc_conf_file
 
     def __getattr__(self, name):
         if name.startswith('__') and name.endswith('__'):
@@ -179,7 +180,9 @@ class SimpleBitcoinProxy:
             raise AttributeError
 
         # Create a callable to do the actual call
-        f = lambda *args: BitcoinProxy(self.url)._call(name, *args)
+        proxy = BitcoinProxy(btc_conf_file=self.__btc_conf_file__)
+
+        f = lambda *args: proxy._call(name, *args)
 
         # Make debuggers show <function bitcoin.rpc.name> rather than <function
         # bitcoin.rpc.<lambda>>
@@ -210,10 +213,10 @@ class BitcoinD(TailableProc):
             '-nolisten',
         ]
         BITCOIND_CONFIG['rpcport'] = rpcport
+        btc_conf_file = os.path.join(regtestdir, 'bitcoin.conf')
         write_config(os.path.join(bitcoin_dir, 'bitcoin.conf'), BITCOIND_CONFIG)
-        write_config(os.path.join(regtestdir, 'bitcoin.conf'), BITCOIND_CONFIG)
-        self.rpc = SimpleBitcoinProxy(
-            "http://rpcuser:rpcpass@127.0.0.1:{}".format(self.rpcport))
+        write_config(btc_conf_file, BITCOIND_CONFIG)
+        self.rpc = SimpleBitcoinProxy(btc_conf_file=btc_conf_file)
 
     def start(self):
         TailableProc.start(self)

